priam/src/main/java/com/netflix/priam/restore/AbstractRestore.java

```diff
@@ -95,161 +89,94 @@ public AbstractRestore(
-        if (waitForCompletion) waitForCompletion(futureList);
-
         return futureList;
     }
 
     private void waitForCompletion(List<Future<Path>> futureList) throws Exception {
         for (Future<Path> future : futureList) future.get();
     }
 
-    private void stopCassProcess() throws IOException {
-        cassProcess.stop(true);
-    }
-
     @Override
     public void execute() throws Exception {
-        if (!isRestoreEnabled(config, instanceIdentity.getInstanceInfo())) return;
+        if (!isRestoreEnabled(config)) return;
 
         logger.info("Starting restore for {}", config.getRestoreSnapshot());
         final DateUtil.DateRange dateRange = new DateUtil.DateRange(config.getRestoreSnapshot());
         new RetryableCallable<Void>() {
             public Void retriableCall() throws Exception {
-                logger.info("Attempting restore");
                 restore(dateRange);
-                logger.info("Restore completed");
-
                 // Wait for other server init to complete
                 sleeper.sleep(30000);
                 return null;
             }
         }.call();
+        logger.info("Restore complete.");
     }
 
     public void restore(DateUtil.DateRange dateRange) throws Exception {
-        // fail early if post restore hook has invalid parameters
         if (!postRestoreHook.hasValidParameters()) {
             throw new PostRestoreHookException("Invalid PostRestoreHook parameters");
         }
-
-        Date endTime = new Date(dateRange.getEndTime().toEpochMilli());
         IMetaProxy metaProxy = metaV2Proxy;
-
-        // Set the restore status.
-        instanceState.getRestoreStatus().resetStatus();
-        instanceState
-                .getRestoreStatus()
-                .setStartDateRange(
-                        LocalDateTime.ofInstant(dateRange.getStartTime(), ZoneId.of("UTC")));
-        instanceState.getRestoreStatus().setEndDateRange(DateUtil.convert(endTime));
-        instanceState.getRestoreStatus().setExecutionStartTime(LocalDateTime.now());
-        instanceState.setRestoreStatus(Status.STARTED);
+        instanceState.startRestore(dateRange);
         String origToken = instanceIdentity.getInstance().getToken();
-
         try {
             if (config.isRestoreClosestToken()) {
-                restoreToken =
+                BigInteger restoreToken =
                         tokenSelector.getClosestToken(
                                 new BigInteger(origToken),
                                 new Date(dateRange.getStartTime().toEpochMilli()));
                 instanceIdentity.getInstance().setToken(restoreToken.toString());
             }
-
-            // Stop cassandra if its running
-            stopCassProcess();
-
-            // Cleanup local data
-            File dataDir = new File(config.getDataFileLocation());
-            if (dataDir.exists() && dataDir.isDirectory()) FileUtils.cleanDirectory(dataDir);

```

Comment:
Question about this: I know we don't need to stop `cass` anymore because we use `import` but I was surprised to see we no longer clean the directory. Is this for the case where we want to restore a subset of the data but keep whats there? Do we still need the option to clean the directory for some restores?